api/screenshot/cache.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package screenshot
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"cloud.google.com/go/storage"
+
+	"github.com/web-platform-tests/wpt.fyi/shared"
+)
+
+func getHashesHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := shared.NewAppEngineContext(r)
+	ds := shared.NewAppEngineDatastore(ctx, false)
+
+	browser := r.FormValue("browser")
+	browserVersion := r.FormValue("browser_version")
+	os := r.FormValue("os")
+	osVersion := r.FormValue("os_version")
```

Comment:
Why the separate params? This could just be a product(s) param, and leverage the existing syntax + parsing for, say, `chrome-72-linux` type thing?

Reply:
This is somewhat complicated: the request will come from `wptrunner`, where browser names and versions are not normalized (the normalization happens in the processor), so we'd see invalid versions (e.g. Safari TP) if we use `ParseProduct`.